### ISSUE TITLE
Run shellcheck manually and fix all warnings on Wazuh installation assistant

### ISSUE THREAD

**Issue by alberpilot** [wazuh/wazuh-packages#1572]
It's needed to run Shellsheck to verify all warnings of the installation assistant and clean it.

**Comment by miguelfdez99** [comment]
Running wazuh-install.sh in shellcheck gives many warnings and information about the script.  There is a lot of code that can be improved.
The most common shellcheck output is [SC2086](https://www.shellcheck.net/wiki/SC2086) which means using double quotes to prevent globbing and word splitting.
Here's a list with other usual warnings and info:
- [SC2235](https://www.shellcheck.net/wiki/SC2235): **Use { ..; } instead of (..) to avoid subshell overhead.**
This particular issue was repeated a lot. Example:

````
if [ -z "${overwrite}" ] && ([ -n "${wazuh_installed}" ] || [ -n "${wazuh_remaining_files}" ]); then
````
This code can be transformed into the following code:

````
if  [ -z "${overwrite}" ] && { [ -n "${wazuh_installed}" ] || [ -n "${wazuh_remaining_files}" ]; }; then
````
- [ SC2234](https://www.shellcheck.net/wiki/SC2234): **Remove superfluous (..) around test command to avoid subshell overhead.**
Wrapping a single test command creates an unnecessary subshell, making it slower.

````
 if ([ "${DIST_NAME}" == "amzn" ]) && ([ "${DIST_VER}" -ne "2" ]); then
````
````
 if [ "${DIST_NAME}" == "amzn" ] && [ "${DIST_VER}" -ne "2" ]; then
````

- [SC2206](https://www.shellcheck.net/wiki/SC2206): **Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.**
If the array has a single element we can quote it. If we want to split it into words or lines we have to use mapfile, read -a, or while.
Problematic code:

````
fileusers=(${sfileusers})
````
This will invoke the shell’s sloppy word splitting and glob expansion.
Correct code:

````
mapfile -t fileusers <<< "${sfileusers}"
````
-[SC2002](https://www.shellcheck.net/wiki/SC2002): **Useless cat. Consider cmd < file | .. or cmd file | .. instead.**
Wrong code:
````
cores=$(cat /proc/cpuinfo | grep -c processor )
````
Correct code:
````
cores=$( grep -c processor /proc/cpuinfo )
````
- [SC2129](https://www.shellcheck.net/wiki/SC2129): **Consider using { cmd1; cmd2; } >> file instead of individual redirects.**
Rather than adding >> something after every single line, you can simply group the relevant commands and redirect the group. So the file has to be opened and closed only once and it means a performance gain.
Problematic code:
````
echo "# ${user_description[${i}]}" >> "${gen_file}"
echo "  username: ${users[${i}]}" >> "${gen_file}"
echo "  password: ${passwords[${i}]}" >> "${gen_file}"
echo ""	>> "${gen_file}"
````
Correct code:
````
{
echo "# ${user_description[${i}]}"
echo "  username: ${users[${i}]}"
echo "  password: ${passwords[${i}]}" 
echo ""	
} >> "${gen_file}"
````

**Comment by miguelfdez99** [comment]
# Tests
## wazuh-certs-tool.sh

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 11:02:08 INFO: Admin certificates created.
30/06/2022 11:02:08 INFO: Wazuh indexer certificates created.
30/06/2022 11:02:08 INFO: Wazuh server certificates created.
30/06/2022 11:02:08 INFO: Wazuh dashboard certificates created.

[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -ca
30/06/2022 11:03:19 INFO: Authority certificates created.
[root@ip-172-31-30-36 ec2-user]# mv wazuh-certificates/ certs/
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -wd certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key 
30/06/2022 11:04:18 INFO: Wazuh dashboard certificates created.
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -wi certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key  
30/06/2022 11:04:57 ERROR: Directory wazuh-certificates already exists in the same path as the script. Please, remove the certs directory to create new certificates.
[root@ip-172-31-30-36 ec2-user]# rm -rf wazuh-certificates/   
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -wi certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key 
30/06/2022 11:05:11 INFO: Wazuh indexer certificates created.
[root@ip-172-31-30-36 ec2-user]# rm -rf wazuh-certificates/
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -ws certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key  
30/06/2022 11:05:21 INFO: Wazuh server certificates created.
[root@ip-172-31-30-36 ec2-user]# rm -rf wazuh-certificates/
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -s certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key 
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -a certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key 
30/06/2022 11:05:48 INFO: Admin certificates created.
````
Errors:

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:48:29 ERROR: Duplicated indexer node names.
[root@ip-172-31-30-36 ec2-user]# nano config.yml         
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:48:42 ERROR: Duplicated indexer node ips.
[root@ip-172-31-30-36 ec2-user]# nano config.yml         
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:48:57 ERROR: Duplicated Wazuh server node names.
[root@ip-172-31-30-36 ec2-user]# nano config.yml         
[root@ip-172-31-30-36 ec2-user]# nano config.yml 
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:49:30 ERROR: Duplicated Wazuh server node ips.
[root@ip-172-31-30-36 ec2-user]# nano config.yml         
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:49:41 ERROR: Duplicated dashboard node ips.
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:53:02 ERROR: Duplicated dashboard node names.

[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:50:04 ERROR: The tag node_type needs to be specified for all Wazuh server nodes.
[root@ip-172-31-30-36 ec2-user]# nano config.yml 
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:50:45 ERROR: Different number of dashboard node names and IPs.
[root@ip-172-31-30-36 ec2-user]# nano config.yml 
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 10:51:02 ERROR: Wazuh cluster needs a single master node.
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -A
30/06/2022 11:01:48 ERROR: The tag node_type can only be used with more than one Wazuh server.
[root@ip-172-31-30-36 ec2-user]# ./wazuh-certs-tool.sh -wi certs/wazuh-certificates/root-ca.pem certs/wazuh-certificates/root-ca.key  
30/06/2022 11:04:57 ERROR: Directory wazuh-certificates already exists in the same path as the script. Please, remove the certs directory to create new certificates.
````
## wazuh-passwords-tool.sh

<details><summary> Passwords file </summary>

````
[root@ip-172-31-30-36 ec2-user]# cat wazuh-install-files/passwords.wazuh 
# Admin user for the web user interface and Wazuh indexer. Use this user to log in to Wazuh dashboard
  indexer_username: 'admin'
  indexer_password: 'h?amD5I9Yp05b0knPK9m+wDFiVhg3*P2'

# Wazuh dashboard user for establishing the connection with Wazuh indexer
  indexer_username: 'kibanaserver'
  indexer_password: 'ZW7s17DKMIH*r+sYyYPac+.8Q6cXzZc0'

# Regular Dashboard user, only has read permissions to all indices and all permissions on the .kibana index
  indexer_username: 'kibanaro'
  indexer_password: 'aPK3Tr*wSe*d5xGJvB.VU4Xa84s0.s8x'

# Filebeat user for CRUD operations on Wazuh indices
  indexer_username: 'logstash'
  indexer_password: '0XwqHAfYaJPy5E3eoJ?QaAdWKFlIKlLR'

# User with READ access to all indices
  indexer_username: 'readall'
  indexer_password: '0+uxZZ602B7Q7RRN*apr+Xm3caVi39c1'

# User with permissions to perform snapshot and restore operations
  indexer_username: 'snapshotrestore'
  indexer_password: 'cJ.kY*+5rsSujT0A.jRmd7lXxfX1L.Zu'

# Admin user used to communicate with Wazuh API
  indexer_username: 'wazuh_admin'
  indexer_password: 'rjZamp5Sq7Yu?tgw6x8ynTTlBjPnKJC+'

# Regular user to query Wazuh API
  indexer_username: 'wazuh_user'
  indexer_password: '1PbcXfL+7xjWMhcjGS7kkNxxOQ*qevFW'

# Password for wazuh API user
  api_username: 'wazuh'
  api_password: 'cVr0mj4Xvn33knRKK?*7seLFC6xX53jr'

# Password for wazuh-wui API user
  api_username: 'wazuh-wui'
  api_password: 'V2mKv0PN.yhqZhbLAPhcK9*M7*DHDA9O'
````


</details>

<details><summary>  -a </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -a
30/06/2022 10:01:47 INFO: Wazuh API admin credentials not provided, Wazuh API passwords not changed.
30/06/2022 10:02:08 INFO: The password for user admin is QYM2QkKcX84wL+2Abu+b?bLKsIxpA+GW
30/06/2022 10:02:08 INFO: The password for user kibanaserver is ATISpWFUUjYRg2w?tEdwbndt6itSt+64
30/06/2022 10:02:08 INFO: The password for user kibanaro is ?BdxBECzAGsIotcAwOSfptd31vcC9Zhx
30/06/2022 10:02:08 INFO: The password for user logstash is docGhN8IHQ7NK80CoA*oXCFm5QhZWls5
30/06/2022 10:02:08 INFO: The password for user readall is ?3.uGce5q1jc2y3GcOzpdsE3DeeSuRna
30/06/2022 10:02:08 INFO: The password for user snapshotrestore is *.7W1TY*GWl?OVc64tGgZDwNscVLxgD5
30/06/2022 10:02:08 INFO: The password for user wazuh_admin is txT6J4.Ox5654?+q3BcZ+MOVoTqYV75U
30/06/2022 10:02:08 INFO: The password for user wazuh_user is vWRl*7K4+A2KbM2Hid3VCQczmwf8He8c
30/06/2022 10:02:08 WARNING: Wazuh indexer passwords changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
````
</details>

<details><summary> -u user </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -u wazuh_user
30/06/2022 10:04:09 INFO: Generating password hash
30/06/2022 10:04:15 INFO: 
The password for user wazuh_user is *5pBR0e+9bKrYGbBuoMRIktOhZfgG+4Z
30/06/2022 10:04:15 WARNING: Password changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
````
</details>

<details><summary>  -u user -p password</summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -u wazuh_user -p *5pBR0e+9bKrYG-RIktOhZfgG+4Z
30/06/2022 10:04:46 INFO: Generating password hash
30/06/2022 10:04:51 WARNING: Password changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
````
</details>

<details><summary> -f file </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -f wazuh-install-files/passwords.wazuh 
30/06/2022 10:10:56 INFO: Wazuh API admin credentials not provided, Wazuh API passwords not changed.
30/06/2022 10:11:17 INFO: The password for user admin is h?amD5I9Yp05b0knPK9m+wDFiVhg3*P2
30/06/2022 10:11:17 INFO: The password for user kibanaserver is ZW7s17DKMIH*r+sYyYPac+.8Q6cXzZc0
30/06/2022 10:11:17 INFO: The password for user kibanaro is aPK3Tr*wSe*d5xGJvB.VU4Xa84s0.s8x
30/06/2022 10:11:17 INFO: The password for user logstash is 0XwqHAfYaJPy5E3eoJ?QaAdWKFlIKlLR
30/06/2022 10:11:17 INFO: The password for user readall is 0+uxZZ602B7Q7RRN*apr+Xm3caVi39c1
30/06/2022 10:11:17 INFO: The password for user snapshotrestore is cJ.kY*+5rsSujT0A.jRmd7lXxfX1L.Zu
30/06/2022 10:11:17 INFO: The password for user wazuh_admin is rjZamp5Sq7Yu?tgw6x8ynTTlBjPnKJC+
30/06/2022 10:11:17 INFO: The password for user wazuh_user is 1PbcXfL+7xjWMhcjGS7kkNxxOQ*qevFW
30/06/2022 10:11:17 WARNING: Wazuh indexer passwords changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
````

</details>

<details><summary> -a -au admin-user -ap admin-password  </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -a -au wazuh -ap cVr0mj4Xvn33knRKK?*7seLFC6xX53jr
30/06/2022 10:14:14 INFO: The password for user admin is PoFswvQ*AwnKyH2oWgkJv1xVfTFOhQaE
30/06/2022 10:14:14 INFO: The password for user kibanaserver is 8.P7x?NyNv2L6R1CE+rM42uTJHy*hNVQ
30/06/2022 10:14:14 INFO: The password for user kibanaro is BnwuhS3nKm5WirDVLJykf..?1QQcUOBI
30/06/2022 10:14:14 INFO: The password for user logstash is esydk4C+BSOTQwauldS*t.FP+CTOI?9T
30/06/2022 10:14:14 INFO: The password for user readall is HnXCM6dMzq??2ikSLc75R12n8ArN.h3m
30/06/2022 10:14:14 INFO: The password for user snapshotrestore is 2+0rPVzP2OeK*sKCzXfHfu2YpJvZhkV+
30/06/2022 10:14:14 INFO: The password for user wazuh_admin is Sl4iWOqaBVnS9Sqji7KvHf+V4EY8T4Am
30/06/2022 10:14:14 INFO: The password for user wazuh_user is um2hZi.R70X93MRAOohj8gAR4vXwe.*p
30/06/2022 10:14:14 WARNING: Wazuh indexer passwords changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
30/06/2022 10:14:15 INFO: The password for Wazuh API user wazuh is ltHbf?RCyMur9*7Tn*36gaZa.lC1Qy6X
30/06/2022 10:14:16 INFO: The password for Wazuh API user wazuh-wui is ?QvRKpyWNDpfLrfut.E1YAnzqmgar9jp
30/06/2022 10:14:16 INFO: Updated wazuh-wui user password in wazuh dashboard. Remember to restart the service.
````
</details>

<details><summary> -f file -au admin-user -ap admin-password </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -f wazuh-install-files/passwords.wazuh -au wazuh -ap ltHbf?RCyMur9*7Tn*36gaZa.lC1Qy6X
30/06/2022 10:25:11 INFO: The password for user admin is h?amD5I9Yp05b0knPK9m+wDFiVhg3*P2
30/06/2022 10:25:11 INFO: The password for user kibanaserver is ZW7s17DKMIH*r+sYyYPac+.8Q6cXzZc0
30/06/2022 10:25:11 INFO: The password for user kibanaro is aPK3Tr*wSe*d5xGJvB.VU4Xa84s0.s8x
30/06/2022 10:25:11 INFO: The password for user logstash is 0XwqHAfYaJPy5E3eoJ?QaAdWKFlIKlLR
30/06/2022 10:25:11 INFO: The password for user readall is 0+uxZZ602B7Q7RRN*apr+Xm3caVi39c1
30/06/2022 10:25:11 INFO: The password for user snapshotrestore is cJ.kY*+5rsSujT0A.jRmd7lXxfX1L.Zu
30/06/2022 10:25:11 INFO: The password for user wazuh_admin is rjZamp5Sq7Yu?tgw6x8ynTTlBjPnKJC+
30/06/2022 10:25:11 INFO: The password for user wazuh_user is 1PbcXfL+7xjWMhcjGS7kkNxxOQ*qevFW
30/06/2022 10:25:11 WARNING: Wazuh indexer passwords changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
30/06/2022 10:25:13 INFO: The password for Wazuh API user wazuh is cVr0mj4Xvn33knRKK?*7seLFC6xX53jr
30/06/2022 10:25:14 INFO: The password for Wazuh API user wazuh-wui is V2mKv0PN.yhqZhbLAPhcK9*M7*DHDA9O
30/06/2022 10:25:14 INFO: Updated wazuh-wui user password in wazuh dashboard. Remember to restart the service.
````
</details>

<details><summary> -au admin-user -ap admin-password -u  </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -a -au wazuh -ap cVr0mj4Xvn33knRKK?*7seLFC6xX53jr
30/06/2022 10:14:14 INFO: The password for user admin is PoFswvQ*AwnKyH2oWgkJv1xVfTFOhQaE
30/06/2022 10:14:14 INFO: The password for user kibanaserver is 8.P7x?NyNv2L6R1CE+rM42uTJHy*hNVQ
30/06/2022 10:14:14 INFO: The password for user kibanaro is BnwuhS3nKm5WirDVLJykf..?1QQcUOBI
30/06/2022 10:14:14 INFO: The password for user logstash is esydk4C+BSOTQwauldS*t.FP+CTOI?9T
30/06/2022 10:14:14 INFO: The password for user readall is HnXCM6dMzq??2ikSLc75R12n8ArN.h3m
30/06/2022 10:14:14 INFO: The password for user snapshotrestore is 2+0rPVzP2OeK*sKCzXfHfu2YpJvZhkV+
30/06/2022 10:14:14 INFO: The password for user wazuh_admin is Sl4iWOqaBVnS9Sqji7KvHf+V4EY8T4Am
30/06/2022 10:14:14 INFO: The password for user wazuh_user is um2hZi.R70X93MRAOohj8gAR4vXwe.*p
30/06/2022 10:14:14 WARNING: Wazuh indexer passwords changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
30/06/2022 10:14:15 INFO: The password for Wazuh API user wazuh is ltHbf?RCyMur9*7Tn*36gaZa.lC1Qy6X
30/06/2022 10:14:16 INFO: The password for Wazuh API user wazuh-wui is ?QvRKpyWNDpfLrfut.E1YAnzqmgar9jp
30/06/2022 10:14:16 INFO: Updated wazuh-wui user password in wazuh dashboard. Remember to restart the service.
````
</details>

<details><summary> -f file -au admin-user -ap admin-password </summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -f wazuh-install-files/passwords.wazuh -au wazuh -ap ltHbf?RCyMur9*7Tn*36gaZa.lC1Qy6X
30/06/2022 10:25:11 INFO: The password for user admin is h?amD5I9Yp05b0knPK9m+wDFiVhg3*P2
30/06/2022 10:25:11 INFO: The password for user kibanaserver is ZW7s17DKMIH*r+sYyYPac+.8Q6cXzZc0
30/06/2022 10:25:11 INFO: The password for user kibanaro is aPK3Tr*wSe*d5xGJvB.VU4Xa84s0.s8x
30/06/2022 10:25:11 INFO: The password for user logstash is 0XwqHAfYaJPy5E3eoJ?QaAdWKFlIKlLR
30/06/2022 10:25:11 INFO: The password for user readall is 0+uxZZ602B7Q7RRN*apr+Xm3caVi39c1
30/06/2022 10:25:11 INFO: The password for user snapshotrestore is cJ.kY*+5rsSujT0A.jRmd7lXxfX1L.Zu
30/06/2022 10:25:11 INFO: The password for user wazuh_admin is rjZamp5Sq7Yu?tgw6x8ynTTlBjPnKJC+
30/06/2022 10:25:11 INFO: The password for user wazuh_user is 1PbcXfL+7xjWMhcjGS7kkNxxOQ*qevFW
30/06/2022 10:25:11 WARNING: Wazuh indexer passwords changed. Remember to update the password in the Wazuh dashboard and Filebeat nodes if necessary, and restart the services.
30/06/2022 10:25:13 INFO: The password for Wazuh API user wazuh is cVr0mj4Xvn33knRKK?*7seLFC6xX53jr
30/06/2022 10:25:14 INFO: The password for Wazuh API user wazuh-wui is V2mKv0PN.yhqZhbLAPhcK9*M7*DHDA9O
30/06/2022 10:25:14 INFO: Updated wazuh-wui user password in wazuh dashboard. Remember to restart the service.
````
</details>

<details><summary>  -au admin-user -ap admin-password -u user -p password -A</summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -au wazuh -ap cVr0mj4Xvn33knRKK?*7seLFC6xX53jr -u wazuh -p BkJt92r*ndzN.CkCYWn?d7i5Z7EaUt63 -A 
30/06/2022 10:29:40 INFO: The password for Wazuh API user wazuh is BkJt92r*ndzN.CkCYWn?d7i5Z7EaUt63
````
</details>

<details><summary>  Expected errors</summary>

````
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -u wazuuuh
30/06/2022 10:34:32 ERROR: The given user does not exist
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -au wazuh -ap cVr0mj4Xvn33knRKK?*7seLFC6xX53jr -u wazuuuh -p BkJt92r*ndzN.CkCYWn?d7i5Z7EaUt -A 
30/06/2022 10:35:04 ERROR: Invalid admin user credentials
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -au wazuh -ap cVr0mj4Xvn33knRKK?*7seLFC6xX53jr -u wazuh -p password -A 
30/06/2022 10:35:37 ERROR: Invalid admin user credentials
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -au wazuh -ap BkJt92r*ndzN.CkCYWn?d7i5Z7EaUt63 -u wazuh -p password -A 
30/06/2022 10:36:04 ERROR: The password must have a length between 8 and 64 characters and contain at least one upper and lower case letter, a number and a symbol(.*+?-).
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -u wazuh -p password
30/06/2022 10:36:34 ERROR: The given user does not exist
[root@ip-172-31-30-36 ec2-user]# ./wazuh-passwords-tool.sh -u wazuh_user -p password
30/06/2022 10:36:43 ERROR: The password must have a length between 8 and 64 characters and contain at least one upper and lower case letter, a number and a symbol(.*+?-).
````
</details>

## wazuh-install.sh
<details><summary> All in one</summary>

````

[root@ip-172-31-30-36 ec2-user]# bash wazuh-install.sh -a
30/06/2022 09:40:55 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.5
30/06/2022 09:40:55 INFO: Verbose logging redirected to /var/log/wazuh-install.log
30/06/2022 09:41:00 INFO: Wazuh repository added.
30/06/2022 09:41:00 INFO: --- Configuration files ---
30/06/2022 09:41:00 INFO: Generating configuration files.
30/06/2022 09:41:01 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
30/06/2022 09:41:01 INFO: --- Wazuh indexer ---
30/06/2022 09:41:01 INFO: Starting Wazuh indexer installation.
30/06/2022 09:41:49 INFO: Wazuh indexer installation finished.
30/06/2022 09:41:49 INFO: Wazuh indexer post-install configuration finished.
30/06/2022 09:41:49 INFO: Starting service wazuh-indexer.
30/06/2022 09:42:04 INFO: wazuh-indexer service started.
30/06/2022 09:42:04 INFO: Initializing Wazuh indexer cluster security settings.
30/06/2022 09:42:11 INFO: Wazuh indexer cluster initialized.
30/06/2022 09:42:11 INFO: --- Wazuh server ---
30/06/2022 09:42:11 INFO: Starting the Wazuh manager installation.
30/06/2022 09:42:29 INFO: Wazuh manager installation finished.
30/06/2022 09:42:29 INFO: Starting service wazuh-manager.
30/06/2022 09:42:44 INFO: wazuh-manager service started.
30/06/2022 09:42:44 INFO: Starting Filebeat installation.
30/06/2022 09:42:52 INFO: Filebeat installation finished.
30/06/2022 09:42:53 INFO: Filebeat post-install configuration finished.
30/06/2022 09:42:53 INFO: Starting service filebeat.
30/06/2022 09:42:53 INFO: filebeat service started.
30/06/2022 09:42:53 INFO: --- Wazuh dashboard ---
30/06/2022 09:42:53 INFO: Starting Wazuh dashboard installation.
30/06/2022 09:43:45 INFO: Wazuh dashboard installation finished.
30/06/2022 09:43:45 INFO: Wazuh dashboard post-install configuration finished.
30/06/2022 09:43:45 INFO: Starting service wazuh-dashboard.
30/06/2022 09:43:45 INFO: wazuh-dashboard service started.
30/06/2022 09:44:19 INFO: Updated wazuh-wui user password in wazuh dashboard. Remember to restart the service.
30/06/2022 09:44:24 INFO: Initializing Wazuh dashboard web application.
30/06/2022 09:44:24 INFO: Wazuh dashboard web application initialized.
30/06/2022 09:44:24 INFO: --- Summary ---
30/06/2022 09:44:24 INFO: You can access the web interface https://<wazuh-dashboard-ip>
    User: admin
    Password: h?amD5I9Yp05b0knPK9m+wDFiVhg3*P2
30/06/2022 09:44:24 INFO: Installation finished.
````

</details>

<details><summary> Distributed </summary>

</details>

**Comment by miguelfdez99** [comment]
# CPU USAGE - AIO

<details><summary> Shellcheck script </summary>

</details>

<details><summary> Current script </summary>

</details>
